File: exercise/ts/week01-code-analysis/src/movie.ts
export class Movie {
    public movieID: string;
    public title: string;
    public director: string;
    public unitPrice: number;
    public totalCopies: number;
    public borrowedCopies: number;

    constructor(movieID: string, title: string, director: string, totalCopies: number, unitPrice: number) {
        this.movieID = movieID;
        this.title = title;
        this.director = director;
        this.totalCopies = totalCopies;
        this.unitPrice = unitPrice;
        this.borrowedCopies = 0;
    }

    canSell(): boolean {
        return this.unitPrice !== 0;
    }
}

File: exercise/kotlin/week01/src/main/kotlin/movie/Movie.kt
package movie

class Movie(
    var movieID: String,
    var title: String,
    var director: String,
    var totalCopies: Int,
    var unitPrice: Double
) {
    var borrowedCopies: Int = 0

    fun canSell(): Boolean {
        return unitPrice != 0.0
    }
}
